lib/pdk/validate/puppet/puppet_environment_conf_validator.rb

```diff
@@ -0,0 +1,92 @@
+require 'pdk'
+
+module PDK
+  module Validate
+    module Puppet
+      class PuppetEnvironmentConfValidator < InternalRubyValidator
+        ALLOWED_SETTINGS = %w[modulepath manifest config_version environment_timeout].freeze
+
+        def name
+          'puppet-environment-conf'
+        end
+
+        def valid_in_context?
+          context.is_a?(PDK::Context::ControlRepo)
+        end
+
+        def pattern
+          ['environment.conf']
+        end
+
+        def spinner_text
+          _('Checking Puppet Environment settings (%{patterns}).') % {
+            patterns: pattern.join(' '),
+          }
+        end
+
+        def validate_target(report, target)
+          unless PDK::Util::Filesystem.readable?(target)
+            report.add_event(
+              file: target,
+              source: name,
+              state: :failure,
+              severity: 'error',
+              message: _('Could not be read.'),
+            )
+            return 1
+          end
+
+          is_valid = true
+          begin
+            env_conf = PDK::ControlRepo.environment_conf_as_config(target)
+
+            env_conf.resolve.each do |setting_name, setting_value|
+              setting_name = setting_name.slice(12..-1)
```

Comment:
I'm guessing this is slicing off an `environment.` prefix? Adding a comment about that or even using `gsub` (performance implications should be pretty minor) would help clarify this for future us.

Reply:
Added a comment.